DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h

```diff
@@ -1,244 +1,286 @@
+                    unsigned int rInv,
                     unsigned int phi0,
                     unsigned int tanl,
                     unsigned int z0,
                     unsigned int d0,
-                    unsigned int Chi2XY,  // would be total chisq if chi2Z is zero
-                    unsigned int Chi2Z,
-                    unsigned int BendChi2,
-                    unsigned int HitPattern,
-                    unsigned int iSpare);
+                    unsigned int chi2RPhi,  // would be total chisq if chi2Z is zero
+                    unsigned int chi2RZ,
+                    unsigned int bendChi2,
+                    unsigned int hitPattern,
+                    unsigned int mvaQuality,
+                    unsigned int mvaOther);
+
+  // ----------copy constructor ----------------------
+  TTTrack_TrackWord(const TTTrack_TrackWord& word) { trackWord_ = word.trackWord_; }
 
-  void setTrackWord(const GlobalVector& Momentum,
+  // ----------operators -----------------------------
+  TTTrack_TrackWord& operator=(const TTTrack_TrackWord& word) {
+    trackWord_ = word.trackWord_;
+    return *this;
+  }
+
+  // ----------member functions (getters) ------------
+  // These functions return arbitarary precision unsigned int words (lists of bits) for each quantity
+  // Signed quantities have the sign enconded in the left-most bit.
+  valid_t getValidWord() const { return getTrackWord()(TrackBitLocations::kValidMSB, TrackBitLocations::kValidLSB); }
+  rinv_t getRinvWord() const { return getTrackWord()(TrackBitLocations::kRinvMSB, TrackBitLocations::kRinvLSB); }
+  phi_t getPhiWord() const { return getTrackWord()(TrackBitLocations::kPhiMSB, TrackBitLocations::kPhiLSB); }
+  tanl_t getTanlWord() const { return getTrackWord()(TrackBitLocations::kTanlMSB, TrackBitLocations::kTanlLSB); }
+  z0_t getZ0Word() const { return getTrackWord()(TrackBitLocations::kZ0MSB, TrackBitLocations::kZ0LSB); }
+  d0_t getD0Word() const { return getTrackWord()(TrackBitLocations::kD0MSB, TrackBitLocations::kD0LSB); }
+  chi2rphi_t getChi2RPhiWord() const {
+    return getTrackWord()(TrackBitLocations::kChi2RPhiMSB, TrackBitLocations::kChi2RPhiLSB);
+  }
+  chi2rz_t getChi2RZWord() const {
+    return getTrackWord()(TrackBitLocations::kChi2RZMSB, TrackBitLocations::kChi2RZLSB);
+  }
+  bendChi2_t getBendChi2Word() const {
+    return getTrackWord()(TrackBitLocations::kBendChi2MSB, TrackBitLocations::kBendChi2LSB);
+  }
+  hit_t getHitPatternWord() const {
+    return getTrackWord()(TrackBitLocations::kHitPatternMSB, TrackBitLocations::kHitPatternLSB);
+  }
+  qualityMVA_t getMVAQualityWord() const {
+    return getTrackWord()(TrackBitLocations::kMVAQualityMSB, TrackBitLocations::kMVAQualityLSB);
+  }
+  otherMVA_t getMVAOtherWord() const {
+    return getTrackWord()(TrackBitLocations::kMVAOtherMSB, TrackBitLocations::kMVAOtherLSB);
+  }
+  tkword_t getTrackWord() const { return tkword_t(trackWord_.to_string().c_str(), 2); }
```

Comment:
Just to note that the `to_string().c_str()` leads to undefined behavior. The lifetime of the temporary string returned by `std::bitset::to_string()` ends at the end of its `c_str()` call, and therefore the `char *` got by constructor of `tkword_t` can point to memory with random content.